src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -162,4 +129,62 @@ public function denormalize($data, $class, $format = null, array $context = arra
 
         return $object;
     }
+
+    /**
+     * Gets and caches attributes for this class and context.
+     *
+     * @param object $object
+     * @param array  $context
+     *
+     * @return array
+     */
+    private function getAttributes($object, array $context)
+    {
+        $key = sprintf('%s-%s', get_class($object), serialize($context));
+
+        if (isset(self::$attributesCache[$key])) {
+            return self::$attributesCache[$key];
+        }
+
+        if (self::$attributesCache[$key] = $this->getAllowedAttributes($object, $context, true)) {
```

Comment:
btw, this means we are missing a test covering the case where the object does not have any property serialized in the groups being used (your PR switches it to serialize the whole object, which is a regression)